cpp/src/arrow/flight/serialization-internal.cc

```diff
@@ -213,9 +232,22 @@ Status FlightDataDeserialize(ByteBuffer* buffer, FlightData* out) {
     switch (field_number) {
       case pb::FlightData::kFlightDescriptorFieldNumber: {
         pb::FlightDescriptor pb_descriptor;
-        if (!pb_descriptor.ParseFromCodedStream(&pb_stream)) {
+        uint32_t length;
+        if (!pb_stream.ReadVarint32(&length)) {
+          return Status(StatusCode::INTERNAL,
+                        "Unable to parse length of FlightDescriptor");
+        }
+        // Can't use ParseFromCodedStream as this reads the entire
+        // rest of the stream into the descriptor command field.
```

Comment:
So we end up having to parse the entire protocol ourselves? Interesting.

Reply:
We're still using the codegen'd `FlightDescriptor::ParseFromString` at least, but it appears we're having to manually pop the descriptor string out of the protocol

Reply:
Yes, by default I found it wanted to parse the entire stream. There's a length-delimited parse method - it might be possible to wrap `GrpcBuffer` in `ArrayInputStream` before wrapping in `CodedInputStream`, I'll give it a try.